src/wallet/mod.rs

```diff
@@ -4728,7 +4771,7 @@ pub(crate) mod test {
         // ...Now the coinbase is mature :)
         let mut builder = wallet.build_tx();
         builder
-            .add_recipient(addr.script_pubkey(), balance / 2)
+            .add_recipient(addr.script_pubkey(), balance.immature / 2)
```

Comment:
Why? At this point the coinbase is mature, the comment above says

Reply:
The coins are mature for the supposed new transaction, but since the database is not updated hence they still are the same. I am using `balance.immature` everywhere because it (the variable) was not updated hence all coins should be in immature category.

I will go ahead and add new tests. Doing everything in one test wouldn't be a good idea, I think.

Reply:
Ok, then it's fine to leave this as is